packages/earljs/test/matchers/ArrayWith.test.ts

```diff
@@ -0,0 +1,57 @@
+import { expect } from 'chai'
+
+import { expect as earlExpect } from '../../src'
+import { ArrayWithMatcher } from '../../src/matchers'
+
+describe('ArrayWith matcher', () => {
+  it('matches a single-element array', () => {
+    const m = new ArrayWithMatcher([1])
+
+    expect(m.check([])).to.be.false
+    expect(m.check(['a'])).to.be.false
+    expect(m.check([1])).to.be.true
+    expect(m.check([1, 2, 3])).to.be.true
+  })
+
+  it('matches a multi-element array', () => {
+    const m = new ArrayWithMatcher([1, 2, 3])
+
+    expect(m.check([])).to.be.false
+    expect(m.check(['a'])).to.be.false
+    expect(m.check([1])).to.be.true
+    expect(m.check([1, 2, 3])).to.be.true
+    expect(m.check([1, 2, 3, 4, 5, 6])).to.be.true
+  })
+
+  it("doesn't match non iterables", () => {
+    const m = new ArrayWithMatcher([1])
+
+    expect(m.check({ something: 1 })).to.be.false
+    expect(m.check(null)).to.be.false
+    expect(m.check(undefined)).to.be.false
+    expect(m.check(1)).to.be.false
+  })
+
+  describe('in expectation', () => {
+    it('works', () => {
+      earlExpect([1, 2, 3]).toEqual(earlExpect.arrayWith(3))
+      earlExpect([1, 2, 3]).toEqual(earlExpect.arrayWith(1, 2, 3))
+      earlExpect([1, 2, 3]).toEqual(earlExpect.arrayWith(1, 2, 3, 4, 5, 6))
```

Comment:
Uhhh I didn't realize this before but I think it should fail in this case. 

I mean when you're expecting that A equals something with X,Y,Z (`arrayWith(X,Y,Z)`) I think it's natural to think that you want all of these elements. 

@sz-piotr thoughts?